### PR TITLE
Customer Home: Change G Suite card to refer to email

### LINKED ISSUE #35928: Customer Home: Replace "Add G suite" card wording with "Get email"
So that we are consistent with /email page.

### PR DESCRIPTION
**Pull request by pablinos**
#### Changes proposed in this Pull Request

It was suggested that the CTA for the G Suite card should be 'Add email'
and that if the customer didn't have a custom domain that we should
point them to that instead. This change implements both of these
changes.

#### Testing instructions

With a site eligible for customer home, view the page. If the site has no custom domain, the CTA will be 'Add a domain' in the last card, linking to `/domains/add`. With a domain, it will be 'Add email', linking to `/email` as it does at the moment.

Fixes #35928

### CODE REVIEW

File: client/my-sites/customer-home/main.jsx [hunk 1]
```diff
@@ -272,14 +274,25 @@ class Home extends Component {
...
-								label={ translate( 'Add G Suite' ) }
-								iconSrc="/calypso/images/customer-home/gsuite.svg"
-							/>
+							{ hasCustomDomain ? (
+								<ActionBox
+									onClick={ () => {
+										trackAction( 'my_site', 'add_gsuite' );
+										page( `/email/${ siteSlug }` );
+									} }
+									label={ translate( 'Add email' ) }
```
Comment (a8ci18n) [inline client/my-sites/customer-home/main.jsx]: Hi! I've found a possible matching string that has already been translated **35** times:
`translate( 'Add Email' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/customer-home/main.jsx [hunk 2]
```diff
@@ -386,6 +399,7 @@ const connectHome = connect(
 			isStaticHomePage: 'page' === getSiteOption( state, siteId, 'show_on_front' ),
 			staticHomePageId: getSiteFrontPage( state, siteId ),
 			showCustomizer: ! isSiteUsingFullSiteEditing( state, siteId ),
+			hasCustomDomain: getDomainsBySiteId( state, siteId ).length > 1,
```
[the remark this answers is not in the document]
Comment (creativecoder) [reply client/my-sites/customer-home/main.jsx]: There's a dead end here, with mapped domains. If you haven't updated the name servers for the domain to point to WordPress.com, you'll get an error message when trying to check out with a G Suite subscription.

Screen Shot 2019-09-18 at 14 25 34

Might it be better to check if there is one site domain that
- `isWPCOMDomain` is false (not a *.wordpress.com domain)
- `pointsToWpcom` is true (the domain is using WordPress.com name servers)
- Not a `*.wpcomstaging.com` domain for an atomic site.

That way the user will be able to successfully purchase email through G Suite.

**Comment by creativecoder** [comment]
Feedback on the testing instructions: it took me some time to figure out how to view the customer home (add user to a/b test and use a newer site). Next time, please include the more generic instructions like you have in #35938 to help fill in reviewers who aren't familiar with this feature. 😀

File: client/my-sites/customer-home/main.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (pablinos) [reply client/my-sites/customer-home/main.jsx]: Nice catch! Yep, sounds like we should do some more checks. Perhaps I'll create a `hasValidDomainForEmail` selector. Seems like that logic might be useful elsewhere.

[the remark this answers is not in the document]
Comment (creativecoder) [reply client/my-sites/customer-home/main.jsx]: And if you want to get this deployed sooner and do that in a separate PR--that would be perfectly reasonable. As is, this still improves the current experience. In that case you have ✅ from me.

**Comment by pablinos** [comment]
I've updated the domain check to use the criteria that's used for the GSuite check. It seems to be working well, and hopefully that function would be updated if the criteria changes.

**Comment by creativecoder** [comment]
The changes look good. The issue with mapped domains that don't yet have nameservers pointing to WordPress.com is still there. You are allowed to proceed through G Suite setup but get an error when adding it to your cart.

However, this now feels like a UX decision regarding G Suite setup, separate from this change. We are giving you a persistent notice about DNS configuration that needs to be fixed, so there's at least an indication that something is wrong--though I'd bet most users wouldn't understand that notice and the email setup error are related. Anyway, I'll open a separate issue, no need to block this from being merged.

File: client/my-sites/customer-home/main.jsx [hunk 3]
```diff
@@ -272,14 +275,25 @@ class Home extends Component {
...
-									trackAction( 'my_site', 'add_gsuite' );
-									page( `/email/${ siteSlug }` );
-								} }
-								label={ translate( 'Add G Suite' ) }
-								iconSrc="/calypso/images/customer-home/gsuite.svg"
-							/>
+							{ hasCustomDomain ? (
+								<ActionBox
+									onClick={ () => {
+										trackAction( 'my_site', 'add_gsuite' );
```
Comment (stephanethomas) [inline client/my-sites/customer-home/main.jsx]: Shouldn't we update this event to `add_email`?

**Pull request merged**